tests/scripts/thread-cert/Cert_8_1_02_Commissioning.py

```diff
@@ -61,12 +65,50 @@ def test(self):
         self.nodes[COMMISSIONER].commissioner_add_joiner(self.nodes[JOINER].get_eui64(), 'PSKD01')
 
         self.nodes[JOINER].interface_up()
-        self.nodes[JOINER].joiner_start('10DKSP')
+        self.nodes[JOINER].joiner_start('PSKD01')
         self.simulator.go(10)
-        self.assertNotEqual(
-            self.nodes[JOINER].get_masterkey(),
-            self.nodes[COMMISSIONER].get_masterkey(),
-        )
+
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        COMMISSIONER = pv.vars['COMMISSIONER']
+        _cpkts = pkts.filter_wpan_src64(COMMISSIONER)
+        _cpkts2 = pkts.filter_wpan_dst64(COMMISSIONER)
+
+        # Step 3: Verify the following details occur in the exchange between
+        # Joiner_1 and the Commissioner
+        # 1. UDP port (Specified by the Commissioner: in Discovery Response) is used as destination port for UDP datagrams from Joiner_1 to the Commissioner.
+        _cpkts.filter_mle_cmd(MLE_DISCOVERY_RESPONSE).must_next().must_verify(
+            lambda p: {
+                EXTENDED_PAN_ID_TLV, NETWORK_NAME_TLV, STEERING_DATA_TLV, COMMISSIONER_UDP_PORT_TLV,
+                JOINER_UDP_PORT_TLV, DISCOVERY_RESPONSE_TLV
+            } == set(p.thread_meshcop.tlv.type))
+
+        # 2. Joiner_1 sends an initial DTLS-ClientHello handshake record to the Commissioner
+        _cpkts2.range(_cpkts.index).filter(lambda p: p.dtls.handshake.type == [HANDSHAKE_CLIENT_HELLO]).must_next()
+
+        # 3. The Commissioner must correctly receive the initial DTLS-ClientHello handshake record and send a DTLS-HelloVerifyRequest handshake record to Joiner_1
+        _pkt = _cpkts.range(
+            _cpkts2.index).filter(lambda p: p.dtls.handshake.type == [HANDSHAKE_HELLO_VERIFY_REQUEST]).must_next()
```

Comment:
check that this message contains non-empty cookie?

Reply:
done